sdks/python/apache_beam/testing/synthetic_pipeline.py

```diff
@@ -520,12 +520,12 @@ def process(
       element,
       restriction_tracker=beam.DoFn.RestrictionParam(
           SyntheticSDFSourceRestrictionProvider())):
-    for k in range(*restriction_tracker.current_restriction()):
-      if not restriction_tracker.try_claim(k):
-        return
-      r = np.random.RandomState(k)
+    cur = restriction_tracker.start_position()
+    while restriction_tracker.try_claim(cur):
```

Comment:
I think a for loop would still be more natural here.

Reply:
Yep, I can change back to for loop.